get_http.go

```diff
@@ -223,6 +266,63 @@ func (g *HttpGetter) parseMeta(r io.Reader) (string, error) {
 	}
 }
 
+// getByteRange is a helper function to parse out the byte range for ranged
+// requests.
+//
+// input values:
+// u must be non-nil and unmodified.
+// example of u:
+// "http://my/file.iso?ranged_request_bytes=5555-66666"
+// "http://my/file.iso?ranged_request_bytes=5555-"
+// note that the format of the raw url string can be either
+// bytes=startByte- OR bytes=startByte-endByte
+//
+// return values:
+// (non-nil, non-nil) - should not occur
+// (nil, nil) - returned when no byte range was provided in the first place
+// (nil, non-nil) - error occured when parsing or validating the byte range
+// (non-nil, nil) - byte range was successfully parsed, and the range is
+// returned as an array of strings in the following format:
+// example non-nil return val "vals":
+// []string{5555, 66666}      // first url example above
+// []string{5555}             // second url example above
+func getByteRange(u *url.URL) ([]string, error) {
+	q := u.Query()
+	byteRange := q.Get("ranged_request_bytes")
+	if byteRange == "" {
+		return nil, nil
+	}
+
+	vals := strings.Split(byteRange, "-")
+
+	// validate byte range values given
+	if len(vals) != 2 {
+		return nil, fmt.Errorf("%s; length of parsed byte range is not 2, %v",
+			invalidRangeMsg, vals)
+	}
+	start, err := strconv.ParseInt(vals[0], 10, 64)
+	if err != nil {
+		return nil, fmt.Errorf("%s; could not convert start byte string to an integer",
+			invalidRangeMsg)
+	}
+	// If you leave endBytes blank, read to end of file.
+	if vals[1] != "" {
+		// otherwise make sure "finish" value is a number
+		finish, err := strconv.ParseInt(vals[1], 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("%s; could not convert finish byte string to an integer",
```

Comment:
Same as above, could you embed the `err` to give the full reason of what went wrong? I like that you are setting context of the error in here though :+1: